### PR TITLE
Various coding guideline fixes

### PR DESCRIPTION
**Pull request by nashif**
﻿- kernel: Make both operands of operators of same essential type category
- kernel/arch: cleanup function definitions
- userspace: rename _is_user_context -> k_is_user_context
- x86: make tests of a value against zero should be made explicit
- kernel: make tests of a value against zero should be made explicit

### CODE REVIEW

File: arch/x86/core/x86_mmu.c [hunk 1]
```diff
@@ -1115,10 +1115,10 @@ static pentry_t flags_to_entry(uint32_t flags)
...
 }
 
-/* map new region virt..virt+size to phys with provided arch-neutral flags */
-void arch_mem_map(void *virt, uintptr_t phys, size_t size, uint32_t flags)
+/* map new region dest..dest+size to addr with provided arch-neutral flags */
+void arch_mem_map(void *dest, uintptr_t addr, size_t size, uint32_t flags)
 {
-	range_map_unlocked(virt, phys, size, flags_to_entry(flags),
+	range_map_unlocked(dest, addr, size, flags_to_entry(flags),
 			   MASK_ALL, 0);
```
Comment (dcpleung) [inline arch/x86/core/x86_mmu.c:1122]: This one is actually better with `phys` and `virt`. Could you change the declaration instead?

File: include/sys/mem_manage.h [hunk 2]
```diff
@@ -270,15 +270,15 @@ void *k_mem_map(size_t size, uint32_t flags);
...
- * @param addr Region base address
+ * @param phys_addr Region base address
  * @param size Region size
  * @param align What to align the address and size to
  * @retval offset between aligned_addr and addr
  */
-size_t k_mem_region_align(uintptr_t *aligned_addr, size_t *aligned_size,
-			  uintptr_t addr, size_t size, size_t align);
+size_t k_mem_region_align(uintptr_t *aligned_phys, size_t *aligned_size,
+			  uintptr_t phys_addr, size_t size, size_t align);
```
Comment (dcpleung) [inline include/sys/mem_manage.h:281]: `addr` and `aligned_addr` are better as it will align any addresses.

File: kernel/device.c [hunk 3]
```diff
@@ -112,7 +112,7 @@ const struct device *z_impl_device_get_binding(const char *name)
 	/* A null string identifies no device.  So does an empty
 	 * string.
 	 */
-	if ((name == NULL) || (*name == 0)) {
+	if ((name == NULL) || (*name == 0U)) {
 		return NULL;
```
Comment (jhedberg) [inline kernel/device.c:116]: Wouldn’t '\0' be the most appropriate here since it’s a character and not an integer?

Comment (jhedberg) [reply kernel/device.c:116]: It would also be more consistent with the place you’re changing in fatal.c

**Review by MaureenHelm — APPROVED** [review]

**Review by enjiamai — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

File: kernel/device.c (hunk 3 above)
Comment (jhedberg) [reply kernel/device.c:116]: Seems like you missed this comment? I think the signedness of `char` is also architecture dependent, which is why `\0` would also be better.

**Review by jhedberg — CHANGES_REQUESTED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**